diff -aurN fbpanel-4.3.orig/Makefile fbpanel-4.3/Makefile
--- fbpanel-4.3.orig/Makefile	2005-05-31 16:55:50.000000000 +0200
+++ fbpanel-4.3/Makefile	2005-06-01 09:44:55.000000000 +0200
@@ -48,8 +48,8 @@
 	$(RM) Makefile.config config.h
 
 install: 
-	install -d $(PREFIX)/bin
-	install -m 755 $(TARGET) $(PREFIX)/bin
+	install -d $(DESTDIR)$(PREFIX)/bin
+	install -m 755 $(TARGET) $(DESTDIR)$(PREFIX)/bin
 
 uninstall:
 	$(RM) $(PREFIX)/bin/$(TARGET)
diff -aurN fbpanel-4.3.orig/Makefile.common fbpanel-4.3/Makefile.common
--- fbpanel-4.3.orig/Makefile.common	2005-05-31 17:39:18.000000000 +0200
+++ fbpanel-4.3/Makefile.common	2005-06-01 09:44:13.000000000 +0200
@@ -89,7 +89,7 @@
 INCS2   = $(shell pkg-config --cflags gtk+-2.0 gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0)
 INCS   := $(INCS2:-I%=-isystem %)
 
-CFLAGS = -O2
+CFLAGS = $(OPTFLAGS)
 ifeq (1,$(DEVEL))
 override CFLAGS += -g
 endif
diff -aurN fbpanel-4.3.orig/config/Makefile fbpanel-4.3/config/Makefile
--- fbpanel-4.3.orig/config/Makefile	2005-05-22 02:46:13.000000000 +0200
+++ fbpanel-4.3/config/Makefile	2005-06-01 09:52:49.000000000 +0200
@@ -17,11 +17,12 @@
 
 
 install: 
-	install -d  $(PREFIX)/share/fbpanel
-	install -m 644 $(TARGET) $(PREFIX)/share/fbpanel
-	install -d  $(PREFIX)/share/fbpanel/images
-	install -m 644 images/*.png $(PREFIX)/share/fbpanel/images
-	install -m 644 images/*.xpm $(PREFIX)/share/fbpanel/images
+	install -d  $(DESTDIR)/etc/fbpanel
+	install -m 644 $(TARGET) $(DESTDIR)/etc/fbpanel
+	install -d  $(DESTDIR)$(PREFIX)/share/fbpanel
+	install -d  $(DESTDIR)$(PREFIX)/share/fbpanel/images
+	install -m 644 images/*.png $(DESTDIR)$(PREFIX)/share/fbpanel/images
+	install -m 644 images/*.xpm $(DESTDIR)$(PREFIX)/share/fbpanel/images
 
 uninstall:
 	$(RM) -r $(PREFIX)/share/fbpanel/images/
diff -aurN fbpanel-4.3.orig/man/Makefile fbpanel-4.3/man/Makefile
--- fbpanel-4.3.orig/man/Makefile	2005-05-22 02:51:08.000000000 +0200
+++ fbpanel-4.3/man/Makefile	2005-06-01 09:46:59.000000000 +0200
@@ -17,8 +17,8 @@
 
 
 install: all
-	install -d  $(PREFIX)/share/man/man1
-	install -m 644 $(TARGET) $(PREFIX)/share/man/man1
+	install -d $(DESTDIR)$(PREFIX)/share/man/man1
+	install -m 644 $(TARGET) $(DESTDIR)$(PREFIX)/share/man/man1
 
 uninstall:
 	$(RM) $(PREFIX)/share/man/man1/$(TARGET)
diff -aurN fbpanel-4.3.orig/panel.c fbpanel-4.3/panel.c
--- fbpanel-4.3.orig/panel.c	2005-05-08 09:46:40.000000000 +0200
+++ fbpanel-4.3/panel.c	2005-06-01 09:50:58.000000000 +0200
@@ -706,8 +706,8 @@
     //ERR("Can't load %s\n", fname);
     g_free(fname);
     
-    /* check private configuration directory */
-    fname = g_strdup_printf("%s/share/fbpanel/%s", PREFIX, profile);
+    /* check system-wide configuration directory */
+    fname = g_strdup_printf("/etc/fbpanel/%s", profile);
     fp = fopen(fname, "r");
     LOG(LOG_INFO, "   %s %s\n", fname, fp ? "ok" : "no");
     if (fp) {
diff -aurN fbpanel-4.3.orig/plugin.c fbpanel-4.3/plugin.c
--- fbpanel-4.3.orig/plugin.c	2005-05-29 06:02:03.000000000 +0200
+++ fbpanel-4.3/plugin.c	2005-06-01 09:49:53.000000000 +0200
@@ -137,7 +137,7 @@
         LOG(LOG_INFO, "   %s ... %s\n", str->str, m ? "ok" : "no");
         if (!m) {
             DBG("error is %s\n", g_module_error());
-            g_string_printf(str, "%s/share/fbpanel/plugins/%s.so", PREFIX, type);
+            g_string_printf(str, "%s/lib/fbpanel/plugins/%s.so", PREFIX, type);
             m = g_module_open(str->str, G_MODULE_BIND_LAZY);
             LOG(LOG_INFO, "   %s ... %s\n", str->str, m ? "ok" : "no");
             if (!m) {
diff -aurN fbpanel-4.3.orig/plugins/Makefile fbpanel-4.3/plugins/Makefile
--- fbpanel-4.3.orig/plugins/Makefile	2005-05-31 17:45:06.000000000 +0200
+++ fbpanel-4.3/plugins/Makefile	2005-06-01 09:48:12.000000000 +0200
@@ -47,8 +47,8 @@
 
 ifeq (0,$(STATIC_PLUGINS))
 install: 
-	install -d  $(PREFIX)/share/fbpanel/plugins
-	install -m 644 $(SOBJ) $(PREFIX)/share/fbpanel/plugins
+	install -d $(DESTDIR)$(LIBDIR)/fbpanel/plugins
+	install -m 644 $(SOBJ) $(DESTDIR)$(LIBDIR)/fbpanel/plugins
 
 uninstall:
 	$(RM) -r $(PREFIX)/share/fbpanel/plugins/
diff -aurN fbpanel-4.3.orig/systray/Makefile fbpanel-4.3/systray/Makefile
--- fbpanel-4.3.orig/systray/Makefile	2005-05-31 17:31:27.000000000 +0200
+++ fbpanel-4.3/systray/Makefile	2005-06-01 09:49:08.000000000 +0200
@@ -41,8 +41,8 @@
 
 ifeq (0,$(STATIC_PLUGINS))
 install: 
-	install -d  $(PREFIX)/share/fbpanel/plugins
-	install -m 644 $(TARGET) $(PREFIX)/share/fbpanel/plugins
+	install -d $(DESTDIR)$(LIBDIR)/fbpanel/plugins
+	install -m 644 $(TARGET) $(DESTDIR)$(LIBDIR)/fbpanel/plugins
 
 uninstall:
 	$(RM) $(PREFIX)/share/fbpanel/plugins/$(TARGET)
