diff -aurN fbpanel-4.1.orig/plugins/menu.c fbpanel-4.1/plugins/menu.c
--- fbpanel-4.1.orig/plugins/menu.c	2004-12-22 15:51:55.000000000 +0100
+++ fbpanel-4.1/plugins/menu.c	2005-05-11 20:56:05.000000000 +0200
@@ -30,9 +30,12 @@
 "{\n"
 "GtkMenuBar::shadow-type = none\n"
 "GtkMenuBar::internal-padding = 0\n"
-"GtkMenu::vertical-padding = 0\n"
+"GtkMenu::vertical-padding = 1\n"
+"GtkMenu::horizontal-offset = -1\n"
+"GtkMenuItem::horizontal-padding = 4\n"
 "}\n"
 "class 'GtkMenuBar' style 'menubar-style'\n"
+"class 'GtkMenuItem' style 'menubar-style'\n"
 "class 'GtkMenu'    style 'menubar-style'\n";
 /*
 static gchar *transparent_rc = "style 'transparent-style'\n"
